.github/workflows/r.yml

```diff
@@ -87,7 +87,7 @@ jobs:
           pip install $(git rev-parse --show-toplevel)
           H2O_VERSION=$(Rscript -e "print(packageVersion('h2o'))" | grep -Eo '[0-9][0-9.]+')
           # test-keras-model.R fails with tensorflow>=2.13.0
-          pip install xgboost 'tensorflow<2.13.0' "h2o==$H2O_VERSION"
+          pip install xgboost 'tensorflow<2.13.0' "h2o==$H2O_VERSION" "pydantic<3,>=1.0"
```

Comment:
do you know why is R affected?

Reply:
Emm I remembered I saw this test failed because of missing this dependency so I just added it, but yea it's a bit weird. I can remove this and investigate why it fails

Reply:
Root cause is here https://github.com/mlflow/mlflow/actions/runs/12408595797/job/34640562122?pr=14100#step:13:418
tensorflow<2.13.0 introduces typing_extensions<4.6.0, and it's not compatible with pydantic.
`tensorflow<2.13.0` is only defined in this r.yml file though, so the test is broken.